### ISSUE TITLE
Create a dial me web-page for peers to know if they are publicly dialable

### ISSUE THREAD

**Issue by aarshkshah1992** [libp2p/go-libp2p#1027]
We should be able to create a webpage wherein peers enter their peerID and we can then tell them if they are publicly dialable or NOT. If they are, we should also tell them the address we dialled them on.

We do need to think of throttling here as the user dosen't do any work expect for entering a peerID.

cc @jacobheun @daviddias

**Comment by aschmahmann** [comment]
We'd still want some throttling (e.g. by IP and/or by IP of who we dial to) to prevent bots from using us to DoS others.

**Comment by aarshkshah1992** [comment]
@aschmahmann Yup, that's mentioned in the description. Otherwise we can easily get DOSed. IIUC, Filecoin does something similar to test connectivity of Filecoin nodes so I'll check with Jeremy on how they do this.

**Comment by daviddias** [comment]
Our experience running this for the Filecoin Space Race is that it becomes mostly a very useful service, with very little incentive to exploit it, as there are many services online to open TCP connections, do pings, port scans and more. A libp2p "dial my peerId" would just be another.

It is important that in this page, there is more information that just "yay/nay". What I envision is a page that can say:

- Yes/No, I can dial to you in that address
- I then looked into the DHT and found these peer addresses for you
- I tried all of them, these are the ones that worked
- From this, I can see that your node supports e.g. TCP, QUIC, NOISE, DHT, etc

This information can be presented in a nice table

In this page you can also add guides "how to do port forwarding" for example

**Comment by aarshkshah1992** [comment]
ping @jacobheun for thoughts and comments.

**Comment by jacobheun** [comment]
@daviddias do you have a link for the Space Race site that was used?

> It is important that in this page, there is more information that just "yay/nay". What I envision is a page that can say:

I agree with this and think these list mentioned is a solid start, and the guides would be helpful. The first thing that comes to mind for me in terms of UI is the flow you see at some of the SSL checker sites, ala https://www.ssllabs.com/ssltest/.

In terms of getting this built out I think we can solicit some folks to get something spun up in the ipfs-shipyard org, and/or create a bounty for this so we can focus on the NAT traversal work. I'll reach out.

**Comment by daviddias** [comment]
@jacobheun check:

- [Experiencing issues deploying a Filecoin miner? Let us know!](https://docs.google.com/document/d/1_oT4lIxsiZz6EI5pudKlSpSfnf3AkobMbZBMCs2RuXA/edit?userstoinvite=koraussie@gmail.com&ts=5f649a46&actionButton=1#heading=h.1gmxmlbc0i0c)
- https://ping.eu/
- https://calibration.spacerace.filecoin.io/check

**Comment by daviddias** [comment]
Has there been any progress on creating this service? Anything available to beta-test?

**Comment by aarshkshah1992** [comment]
cc @yusefnapora @vyzo .
